### ISSUE TITLE
Multiple empty lines in interpreter when sending empty statement at the end of the buffer 

### ISSUE THREAD

**Issue by xelibrion** [jorgenschaefer/elpy#1591]
# Summary 
`elpy-shell-send-statement-and-step` creates multiple empty `In [xx]: ` lines in iPython interpreter when accidentally submitting at the end of the buffer. 

# Steps to reproduce
Insert the following code in the new buffer and run `elpy-shell-send-statement-and-step` at the end of the buffer (on the line after the last `pass` statement). 
```
def fun(args):
    pass

def fun(args):
    pass

def fun(args):
    pass

def fun(args):
    pass
```

Observe the result: 
```
In [38]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
In [39]: 
```
It also seems number of empty statements grows with number of lines in the buffer. 

# My configuration
## OS
macOS 10.13.6 

## Result of `(elpy-config)`
```
Elpy Configuration

Virtualenv........: None
RPC Python........: 3.7.3 (/Users/dmitry.kryuchkov/anaconda3/envs/docker-airflow/bin/python3)
Interactive Python: ipython (/Users/dmitry.kryuchkov/anaconda3/envs/docker-airflow/bin/ipython)
Emacs.............: 26.2
Elpy..............: 1.29.1
Jedi..............: 0.13.3
Rope..............: 0.14.0
Autopep8..........: 1.4.4
Yapf..............: 0.27.0
Black.............: Not found (19.3b0 available)
Syntax checker....: flake8 (/Users/dmitry.kryuchkov/anaconda3/envs/docker-airflow/bin/flake8)

```
Elpy version: `elpy-20190428.2249`

## Elpy configuration in my init.el
```
(use-package elpy
  :pin melpa
  :defer t
  :commands elpy-enable
  :init
  (with-eval-after-load 'python
    (elpy-enable))
  :bind (:map elpy-mode-map
              ("C-c C-f" . nil)
              ;; fix conflict with smartparens
              ("C-<left>" . nil)
              ("C-<right>" . nil))
  :config
  (remove-hook 'elpy-modules 'elpy-module-flymake)
  (setq elpy-rpc-python-command "python3")
  (setq elpy-rpc-backend "rope")
  (setq elpy-modules '(elpy-module-sane-defaults elpy-module-company elpy-module-eldoc elpy-module-yasnippet))
  (setq elpy-shell-use-project-root nil)
  (setq elpy-shell-echo-output nil))
```

**Comment by galaunay** [comment]
Hi, thanks for reporting this.
Strange behaviour indeed !

It is on my to do list, I will have a look as soon as possible.

**Comment by galaunay** [comment]
Should be fixed by PR #1595.

Please re-open if it is not.
